code/modules/mob/living/carbon/human/species_types/beeperson.dm

```diff
@@ -112,3 +127,36 @@
 	if(istype(weapon, /obj/item/melee/flyswatter))
 		return 10
 	return 1
+
+//stinger attack ability
+/datum/action/innate/stinger_attack
+	name = "stingerattack"
+	desc = "launch your stinger"
+	icon_icon = 'icons/mob/actions/actions_bee.dmi'
+	button_icon_state = "stinger_attack"
+	background_icon_state = "bg_spell"
+	var/stinger_path = /obj/item/hardened_stinger
+
+//using the ability
+/datum/action/innate/stinger_attack/Activate(mob/user = usr)
+	var/mob/living/carbon/C = user
+	if(!iscarbon(user))
+		return
+
+	if(HAS_TRAIT(C, TRAIT_NODISMEMBER))
+		return
+
+	var/obj/item/organ/stinger
+	for(var/org in C.internal_organs)
+		if(istype(org, /obj/item/organ/stinger))
+			stinger = org
+			break
+
+	if(!stinger)
+		to_chat(C, "<span class='notice'>You don't have a tongue to shoot!</span>")
```

Comment:
Copy-pasted code :)
Change "tongue" to stinger.

Reply:
i know. i just pushed without editing all, cause im lazy